dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -384,6 +373,59 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
 		}
 	}
 
+    /**
+     * Fields with suffix _text will be indexed if the property `CREATE_TEXT_INDEX_FIELD_FOR_NON_TEXT_FIELDS` is on
+     * @param contentletMap
+     * @param contentlet
+     * @param contentIdentifier
+     * @param versionInfo
+     * @param contentType
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    private void addTextFieldToMapIfNeeded(final Map<String, Object> contentletMap,
```

Comment:
I think we need to treat methods with `void returns` as a code smell  especially if we are passing in an object that is intended to be mutated under the covers.  This to me is an coding anti-pattern.

A better choice for this method for example  is to return an Optional<Map> that if it is present, gets added to the existing collection.  That allows for simpler testing and a clear understanding of what object is being changed when this method is called.

Reply:
done